lib/ansible/plugins/action/gather_facts.py

```diff
@@ -23,7 +23,7 @@ def _get_module_args(self, fact_module, task_vars):
         # deal with 'setup specific arguments'
         if fact_module not in C._ACTION_SETUP:
             # network facts modules must support gather_subset
-            if self._connection._load_name not in ('network_cli', 'httpapi', 'netconf'):
+            if not self._connection._load_name.endswith(('.network_cli', '.httpapi', '.netconf')):
```

Comment:
The last item in the `self._connection._redirected_names` list is the resolved plugin name, so we could check for fqcn specifically. Looks like these are all in ansible.netcommon. Though if the plugins move again it will break again...

Reply:
yeah, why i want to move to 'properties/attributes' vs hardcoded lists